### PR TITLE
[update] delete events by experiment UID

### PR DESCRIPTION
**Pull request by fewdan**
Signed-off-by: “fewdan” <fewdan@hotmail.com>

### What problem does this PR solve?
When deleting archived experiments, delete the corresponding events and pod records.

### What is changed and how does it work?
Delete the corresponding events and pod records by experiment UID.

### Checklist

Tests

- [ ] Unit test
- [ ] E2E test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No code

Side effects

- [ ] Breaking backward compatibility

Related changes

- [ ] Need to update the documentation

### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by WangXiangUSTC** [comment]
the integration test failed because event 1 is deleted when deleting archive experiment, you can move the test of event before archive. 😃

**Comment by fewdan** [comment]
> the integration test failed because event 1 is deleted when deleting archive experiment, you can move the test of event before archive. 😃

oh, let me take a look.

**Review by WangXiangUSTC — APPROVED** [review]
LGTM

File: pkg/store/event/event.go [hunk 1]
```diff
@@ -428,6 +428,27 @@ func (e *eventStore) DeleteByFinishTime(_ context.Context, ttl time.Duration) er
...
 }
 
+// DeleteByUID deletes events by the uid of the experiment.
+func (e *eventStore) DeleteByUID(_ context.Context, uid string) error {
+	eventList, err := e.ListByUID(context.Background(), uid)
+	if err != nil {
+		return err
+	}
+	for _, et := range eventList {
+		if err := e.db.Model(core.Event{}).Unscoped().Delete(*et).Error; err != nil {
```
Comment (g1eny0ung) [inline pkg/store/event/event.go:438]: I think it's better to batch delete all events after we delete all `PodRecord`.

Comment (fewdan) [reply pkg/store/event/event.go:438]: ok. i will modify it.

**Review by g1eny0ung — APPROVED** [review]
LGTM

**Review by cwen0 — APPROVED** [review]
LGTM

**Pull request merged**